base/src/main/java/com/mindera/skeletoid/widgets/tooltip/PopupTooltip.kt

```diff
@@ -22,47 +22,48 @@ import androidx.annotation.DrawableRes
-        @LayoutRes private var layout: Int,
-        @IdRes private var textViewId: Int,
-        @IdRes private var tooltipBackgroundId: Int,
-        //Tooltip layout
-        private var elevation: Int = DEFAULT_ELEVATION,
-        private var popupWidth: Int = TOOLTIP_WIDTH,
-        //Tooltip behaviour
-        private var indefinite: Boolean = false,
-        private var showDuration: Long = 3500L,
-        private var verticalMargin: Int = 0,
-        private var horizontalMargin: Int = 0,
-        private var horizontalMarginClipped: Int = 0,
-        private var verticalMarginClipped: Int = 0,
-        private var insideTouchDismissible: Boolean = true,
-        private var outsideTouchDismissible: Boolean = true,
-        private var respectScreenBounds: Boolean = false,
-        private var modal: Boolean = true,
-        @StyleRes private var animation: Int? = null,
-        //Tooltip arrow
-        private var arrowWidth: Int = ARROW_WIDTH,
-        private var arrowHeight: Int = ARROW_HEIGHT,
-        private var arrow: Boolean = false,
-        @DrawableRes private var arrowDrawable: Int? = null,
-        @ColorRes private var arrowStrokeColor: Int = android.R.color.white,
-        @IdRes private var arrowBackgroundId: Int? = null,
-        @IdRes private var arrowStrokeId: Int? = null
+    private var anchorView: View,
+    private var message: CharSequence,
+    @LayoutRes private var layout: Int,
+    @IdRes private var textViewId: Int,
+    @IdRes private var tooltipBackgroundId: Int
 ) : PopupWindow() {
 
-    companion object {
-
-        /**
-         * The LOGGING TAG. Epic.
-         */
-        const val TAG = "PopupTooltip"
+    /**
+     * The possible orientation for the tooltip in relation to the [anchorView].
+     * Currently it can only be displayed on top or below a view.
+     */
+    enum class TooltipGravity { TOP, BOTTOM, BOTTOM_RIGHT }
+
+    private var elevation: Int = DEFAULT_ELEVATION
+    private var popupWidth: Int = TOOLTIP_WIDTH
+    private var popupHeight: Int = TOOLTIP_HEIGHT
+    private var tooltipGravity: TooltipGravity = TooltipGravity.TOP
+
+    private var indefinite: Boolean = false
+    private var showDuration: Long = 3500L
+    private var verticalMargin: Int = 0
+    private var horizontalMargin: Int = 0
+    private var horizontalMarginClipped: Int = 0
+    private var verticalMarginClipped: Int = 0
+    private var insideTouchDismissible: Boolean = true
+    private var outsideTouchDismissible: Boolean = true
+    private var modal: Boolean = true
+    @StyleRes private var animation: Int? = null
+
+    private var arrowWidth: Int = ARROW_WIDTH
+    private var arrowHeight: Int = ARROW_HEIGHT
+    private var arrow: Boolean = false
+    @DrawableRes private var arrowDrawable: Int? = null
+    @ColorRes private var arrowStrokeColor: Int = android.R.color.white
+    @IdRes private var arrowBackgroundId: Int? = null
+    @IdRes private var arrowStrokeId: Int? = null
```

Comment:
Can we have a builder pattern? Having all these mutable variables is giving me an itch